python/tvm/testing/utils.py

```diff
@@ -981,6 +982,38 @@ def _corstone300_compile_time_check():
 requires_vitis_ai = Feature("vitis_ai", "Vitis AI", cmake_flag="USE_VITIS_AI")
 
 
+def _arm_dot_supported():
+    arch = platform.machine()
+
+    if arch not in ["arm64", "aarch64"]:
+        return False
+
+    if sys.platform.startswith("darwin"):
+        cpu_info = subprocess.check_output("sysctl -a", shell=True).strip().decode()
+        for line in cpu_info.split("\n"):
+            if line.startswith("hw.optional.arm.FEAT_DotProd"):
+                return bool(int(line.split(":", 1)[1]))
+    elif sys.platform.startswith("linux"):
+        return True
+
+    return False
+
+
+def _has_vnni():
+    arch = platform.machine()
+    # Only linux is supported for now.
+    if arch == "x86_64" and sys.platform.startswith("linux"):
+        return "avx512_vnni" in open("/proc/cpuinfo", "r").read()
+
+    return False
+
+
+requires_arm_dot = Feature("arm_dot", "ARM dot product", run_time_check=_arm_dot_supported)
+
+
+requires_cascadelake = Feature("cascadelake", "x86 CascadeLake", run_time_check=_has_vnni)
```

Comment:
Zen4 supports vnni also. Maybe we need another check to make sure the platform is intel.

Reply:
Interesting... I'll just rename this to `requires_vnni`, then. There shouldn't be any difference in Intel vs AMD vnni.

Reply:
The tests use a cascade lake target though.

Reply:
For now, yes. I feel that having both `requires_cascadelake` and `requires_zen4` is redundant. Enabling vnni for zen4 would require minor change in topi x86. So when we make such change, we can parameterize the target string in the test.

Reply:
hmm, even if we parameterize the target string to have both `-mcpu=cascadelake` and `-mcpu=zen4`, we can't run both targets on the same environment... I should stick to `requires_cascadelake`, then.